pmm-app/src/pmm-dbaas/components/DBCluster/EditDBClusterModal/EditDBClusterModal.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { FC, useEffect, useState } from 'react';
+import { Modal } from 'shared/components/Elements/Modal/Modal';
+import { Form as FormFinal } from 'react-final-form';
+import { EditDBClusterFields, EditDBClusterModalProps } from './EditDBClusterModal.types';
+import { DBClusterAdvancedOptions } from './DBClusterAdvancedOptions/DBClusterAdvancedOptions';
+import { DEFAULT_SIZES } from './DBClusterAdvancedOptions/DBClusterAdvancedOptions.constants';
+import { DBClusterTopology } from './DBClusterAdvancedOptions/DBClusterAdvancedOptions.types';
+import { DBClusterServiceFactory } from '../DBClusterService.factory';
+
+export const EditDBClusterModal: FC<EditDBClusterModalProps> = ({
+  isVisible,
+  setVisible,
+  onDBClusterChanged,
+  selectedCluster,
+}) => {
+  const onSubmit = async ({
+    databaseType,
+    topology,
+    nodes,
+    single,
+    memory,
+    cpu,
+    disk,
+  }: Record<string, any>) => {
+    if (!selectedCluster) {
+      setVisible(false);
+
+      return;
+    }
+
+    try {
+      const dbClusterService = DBClusterServiceFactory.newDBClusterService(databaseType.value);
+
+      await dbClusterService.updateDBCluster({
+        databaseType,
+        clusterName: selectedCluster.clusterName,
+        kubernetesClusterName: selectedCluster.kubernetesClusterName,
+        clusterSize: topology === DBClusterTopology.cluster ? nodes : single,
+        cpu,
+        memory,
+        disk,
+      });
+      setVisible(false);
+      onDBClusterChanged();
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  const [initialValues, setInitialValues] = useState({});
+
+  useEffect(() => {
+    if (!selectedCluster) {
```

Comment:
Does this check need to be repeated twice in this component?

Reply:
yep, this is a workaround for TS, I don't know if there's a simpler way to do it